dali/operators/reader/loader/video/video_test_base.cc

```diff
@@ -155,4 +155,18 @@ void VideoTestBase::RunFailureTest(std::function<void()> body, std::string expec
   }
 }
 
+std::vector<char> VideoTestBase::MemoryVideo(std::string &path) {
+  std::ifstream video_file(path, std::ios::binary | std::ios::ate);
+  auto size = video_file.tellg();
+  video_file.seekg(0, std::ios::beg);
+
+  std::vector<char> memory_video(size);
+  if (!video_file.read(memory_video.data(), size)) {
+    // We can't use FAIL() because this function returns value
+    throw ::testing::AssertionFailure() << "Could not load video file to memory.";
+  }
+
+  return memory_video;
```

Comment:
About the return this should be handled by rvo, right? That is why I did it like this. Perf should be the same and it reads clearer.
About the error checking: since setup of this test is non trivial and might fail I wanted to make sure that it worked. So this is just to make sure that the test is ok.
If the file is incomplete `FramesDecoder` will report it properly, with the error from FFMPEG that might say something about this.